### PR TITLE
Send less data for certain user calls to improve data privacy

### PR DESCRIPTION
**Pull request by krusche**
Some REST calls related to users (which are only possible for instructors) reveal more information than needed for the Artemis user interface. This PR tries to reduce the data.

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I updated multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Steps for Testing
1. Add / remove users in the different course groups (students, tutors, instructors)
2. Register single / multiple users for exams
3. Add users to organizations
4. Search for users in those two places

Checks:
* Inspect the REST responses in the chrome developer console and check if unnecessary data (not shown in the user interface, such as groups, authorities, etc.) are not included in the json body.

### CODE REVIEW

**Review by katjanakosic — APPROVED** [review]
Everything worked fine on TS1. Tested with @real-dmy

File: src/main/webapp/app/exam/manage/students/exam-students.component.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ import { iconsAsHTML } from 'app/utils/icons.utils';
 import { Exam } from 'app/entities/exam.model';
 import { ExamManagementService } from 'app/exam/manage/exam-management.service';
 import { ButtonSize, ButtonType } from 'app/shared/components/button.component';
+import { StudentDTO } from 'app/entities/student-dto.model';
```
Comment (FrankeLukas) [inline src/main/webapp/app/exam/manage/students/exam-students.component.ts:16]: This import is unused.

**Review by FrankeLukas — APPROVED** [review]
Adding and removing users from courses, organisations and exams still work (including the check that you can't register instructors for the exam). The searches work, too.

**Review by maximilianbergmann — DISMISSED** [review]
Tested it. Looks good to me. Adding and removing of students worked also fine

**Review by PatZintel — CHANGES_REQUESTED** [review]
When testing on ts2, i can still see the "createdBy"-attribute in the JSON Response even though it is not shown in UI. Also, when adding multiple users to an exam and then getting the list of registered students, i can see the "lastNotificationRead"-attribute of every student. Especially the later one might be a problem for privacy of the individual students. (It might still be needed for the client, i dont know the execution well enough to evaluate that)
Screenshot_createdBy_is_Shown
Screenshot_lastNotificationRead_is_Shown

**Review by real-dmy — APPROVED** [review]
Tested it with @katjanakosic on TS1. Everything worked fine.

**Review by ivanchimeno — APPROVED** [review]
Works as advertised and code looks good

**Review by SimonEntholzer — APPROVED** [review]
Tested on TS1 in testing session :+1:

**Review by ArtemTsurikov — APPROVED** [review]
Tested with @katjanakosic on ts1. Looks good.

**Comment by krusche** [comment]
> When testing on ts2, i can still see the "createdBy"-attribute in the JSON Response even though it is not shown in UI. Also, when adding multiple users to an exam and then getting the list of registered students, i can see the "lastNotificationRead"-attribute of every student. Especially the later one might be a problem for privacy of the individual students. (It might still be needed for the client, i dont know the execution well enough to evaluate that)
> Screenshot_createdBy_is_Shown
> Screenshot_lastNotificationRead_is_Shown

Thanks, both values could be removed, but are less of a privacy issue than the other values. As we already have a couple of approvals, I would suggest that we remove those less problematic values in a follow-up PR

**Review by PatZintel — APPROVED** [review]

**Pull request merged**